app/models/learning.rb

```diff
@@ -16,6 +16,5 @@ class Learning < ActiveRecord::Base
 
 	enum status: [:active, :pending, :progress, :done, :archived]
 
-	validates :status, presence: true
-
+	validates_uniqueness_of :user_id, :course_id 
```

Comment:
I thing these fields should be unique together. Read about uniqueness scope http://guides.rubyonrails.org/active_record_validations.html#uniqueness